raphtory/src/vectors/vectorised_graph.rs

```diff
@@ -56,6 +78,28 @@ impl<G: StaticGraphViewOps> VectorisedGraph<G> {
         Ok(())
     }
 
+    /// Generates and stores embeddings for a batch of edges
+    pub async fn update_edges<T: AsNodeRef>(&self, edges: Vec<(T, T)>) -> GraphResult<()> {
+        let (ids, docs): (Vec<_>, Vec<_>) = edges.iter()
+            .filter_map(|(src, dst)| {
+                self.source_graph.edge(src, dst)
+                    .and_then(|edge| {
+                        let id = edge.edge.pid().0;
+
+                        self.template.edge(edge).map(|doc| (id, doc))
+                    })
+            })
+            .unzip();
+
+        let vectors = self.cache.get_embeddings(docs).await?;
+
+        for (id, vector) in ids.iter().zip(vectors) {
+            self.edge_db.insert_vector(*id, &vector)?;
```

Comment:
Every time you call `insert_vector` this is going to update the index. We should probably change that function to accept a vector of embeddings instead, so we can write everything to arroy and only in the end update the index just once. The same applies to nodes